transmart-core-db/grails-app/services/org/transmartproject/db/userqueries/UserQuerySetService.groovy

```diff
@@ -166,19 +182,19 @@ class UserQuerySetService implements UserQuerySetResource {
     }
 
     private List<QuerySetInstance> getInstancesForLatestQuerySet(Long id) {
-        DetachedCriteria recentDate = DetachedCriteria.forClass(QuerySet)
+        DetachedCriteria criteria = DetachedCriteria.forClass(QuerySet)
                 .add(Restrictions.eq('query.id', id))
-                .setProjection(Projections.max("createDate"))
-
-        // get the content for the most recent createDate and queryId
-        def session = sessionFactory.currentSession
-        QuerySet recent = session.createCriteria(QuerySet)
-                .add(Restrictions.eq("query.id", id))
-                .add(Property.forName( "createDate" ).eq(recentDate))
-                .uniqueResult()
+                .addOrder(Order.desc('createDate'))
```

Comment:
Why is ordering better then projection for the max 'createDate'?

Reply:
Because you can do it in one query instead of two. First fetching the one with max date does a query on the query table, fetching the one with that date also. This solution does fetching the query and selecting the one with the max date in one query. It still only fetches the first, because of the `maxResults`.